Exception: 'NoneType' object has no attribute 'accepted'. Probably requested wrong task name. Available tasks: dict_keys(['search_pipeline', 'inspect_pipeline'])

```
[task_bridge_metacontrol-11] [ERROR] [1701250113.884423592] [adpatation_goal_bridge]: Exception: 'NoneType' object has no attribute 'accepted'. Probably requested wrong task name. Available tasks: dict_keys(['search_pipeline', 'inspect_pipeline'])
1701250337.8803065 [time_constrained_mission-1] [INFO] [1701250337.879468077] [mission_node]: Task inspect_pipeline aborted
1701250337.8825326 [task_bridge_metacontrol-11] [INFO] [1701250337.881849448] [adpatation_goal_bridge]: cancel f_maintain_motion
1701250337.8827627 [task_bridge_metacontrol-11] [ERROR] [1701250337.882242956] [adpatation_goal_bridge]: Exception: 'NoneType' object has no attribute 'cancel_goal_async'. Probably requested wrong task name. Available tasks: dict_keys(['search_pipeline', 'inspect_pipeline'])
```

--- a/suave.rosinstall
+++ b/suave.rosinstall
@@ -30,7 +30,7 @@ repositories:
   mc_mros_reasoner:
     type: git
     url: https://github.com/meta-control/mc_mros_reasoner.git
-    version: 9392f680fb33ed97f6105830dbb5593e68070419
+    version: 9b1f915
   system_modes:
     type: git
     url: https://github.com/micro-ROS/system_modes

--- a/suave/suave/task_bridge.py
+++ b/suave/suave/task_bridge.py
@@ -27,6 +27,16 @@ class TaskBridge(Node):
 
     def task_request(self, req, forward_request):
         response = Task.Response()
+        if req.task_name not in self.task_functions_dict:
+            self.get_logger().error(
+                'Request: {}'.format(req) +
+                'Task requested is not available. ' +
+                'Available tasks: {}'.format(self.task_functions_dict.keys())
+            )
+            response = Task.Response()
+            response.success = False
+            return response
+
         try:
             function_names = self.task_functions_dict[req.task_name]
             success = True
@@ -35,9 +45,7 @@ class TaskBridge(Node):
             response.success = success
         except Exception as e:
             self.get_logger().error(
-                'Exception: {}. '.format(e) +
-                'Probably requested wrong task name. ' +
-                'Available tasks: {}'.format(self.task_functions_dict.keys())
+                'Exception: {0}. Request: {1}'.format(e, req)
             )
             response = Task.Response()
             response.success = False

--- a/suave_metacontrol/launch/metacontrol.launch.py
+++ b/suave_metacontrol/launch/metacontrol.launch.py
@@ -6,8 +6,6 @@ from launch import LaunchDescription
 from launch.actions import IncludeLaunchDescription
 from launch.launch_description_sources import PythonLaunchDescriptionSource
 
-from launch_ros.actions import Node
-
 
 def generate_launch_description():
 

--- a/suave_metacontrol/suave_metacontrol/suave_reasoner.py
+++ b/suave_metacontrol/suave_metacontrol/suave_reasoner.py
@@ -35,7 +35,7 @@ class SuaveReasoner(RosReasoner):
                                 objective.o_status = "IN_ERROR_NFR"
         except Exception as err:
             self.logger.info(
-                "In Custom Analyze, exception returned: {}".format(err))
+                "In custom Analyze, exception returned: {}".format(err))
         return objectives_in_error
 
 
@@ -54,7 +54,7 @@ def main(args=None):
 
     # Spin until the process in terminated
     rclpy.spin(pipeline_inspection_reasoner, executor=mt_executor)
-    ros_reasoner.destroy_node()
+    pipeline_inspection_reasoner.destroy_node()
     rclpy.shutdown()
 
 

--- a/suave_metacontrol/suave_metacontrol/task_bridge_metacontrol.py
+++ b/suave_metacontrol/suave_metacontrol/task_bridge_metacontrol.py
@@ -2,15 +2,12 @@
 import sys
 import rclpy
 
-from diagnostic_msgs.msg import DiagnosticArray
-from diagnostic_msgs.msg import DiagnosticStatus
 from diagnostic_msgs.msg import KeyValue
 from mros2_msgs.action import ControlQos
 
 from rclpy.executors import MultiThreadedExecutor
 from rclpy.callback_groups import MutuallyExclusiveCallbackGroup
 from rclpy.action import ActionClient
-from rclpy.node import Node
 from suave.task_bridge import TaskBridge
 
 

--- a/suave_metacontrol/suave_metacontrol/task_bridge_metacontrol.py
+++ b/suave_metacontrol/suave_metacontrol/task_bridge_metacontrol.py
@@ -29,9 +29,12 @@ class TaskBridgeMetacontrol(TaskBridge):
     def forward_task_request(self, function):
         self.get_logger().info("Waiting for future to complete")
         future = self.send_mros_objective(function)
-        while self.executor.spin_until_future_complete(
-                future, timeout_sec=1.0):
-            self.get_logger().info("Waiting for future to complete")
+
+        self.executor.spin_until_future_complete(future, timeout_sec=5.0)
+        if future.done() is False:
+            self.get_logger().warning(
+                'Future send mros_objective not completed {}'.format(function))
+            return None
         self.current_objectives_handle[function] = future.result()
         return self.current_objectives_handle[function].accepted
 
@@ -41,14 +44,16 @@ class TaskBridgeMetacontrol(TaskBridge):
             goal_handle = self.current_objectives_handle[function]
             future = goal_handle.cancel_goal_async()
             self.get_logger().info('cancel requested {}'.format(function))
-            while self.executor.spin_until_future_complete(
-                    future, timeout_sec=1.0):
-                self.get_logger().info("Waiting for future to complete")
+            self.executor.spin_until_future_complete(future, timeout_sec=5.0)
+            if future.done() is False:
+                self.get_logger().warning(
+                    'Future cancel mros_objective not completed {}'.format(
+                        function))
+                return False
             del function
             if future is not None and goal_handle.status == 3:
                 return True
-            else:
-                return False
+            return False
         else:
             self.get_logger().info(
                 'Function {} is not active'.format(function))

